### ISSUE TITLE
LabVIEW starter package

### ISSUE THREAD

**Issue by amstan** [aichallenge/aichallenge#32]
[CPeng](https://github.com/CPeng) said about 2 hours ago:

> I am new to this whole github thing but I would like to create a starter package for LabVIEW. I was disappointed last challenge when there wasn't one.
> 
> LabVIEW is a graphical dataflow language unlike most text based OO ones. Does it have a place in a competition like this?

Is there a simple, and free way to run this under our linux setup?

Ideally we want a small program that we give the source file to, and it executes it for us. We're not going to be running IDEs for each bot. Our server's don't even have graphical interfaces.

**Comment by McLeopold** [comment]
This may be helpful.
http://joule.ni.com/nidu/cds/view/p/id/1020/lang/en

**Comment by CPeng** [comment]
Thanks for the help guys. LabVIEW is a bit different from the other imperative languages used here but I hope we can work it in.

I've created an application that can read uncompiled LabVIEW source files and run them. It's a 30k DLL linked to from a C file. I guess the next step is to test it on the server.

I would use the newest runtime which is here:
http://joule.ni.com/nidu/cds/view/p/id/2143/lang/en

I will upload my test app as soon as I figure out this git stuff. Is there a good place to put it?

**Comment by amstan** [comment]
> I've created an application that can read uncompiled LabVIEW source files and run them. It's a 30k DLL linked to from a C file. I guess the next step is to test it on the server.

If it's a DLL it will most probably not work under linux. DLLs are windows only, no other platform uses it.

All we need is a linux compiler, we already have the runtime(from that it seems from that link).

You might also want to look into licensing. If we're allowed to freely run this without breaking any copyrights. All compiler/runtime that we used till(including the C#(we used mono instead of the microsoft one)) now were open source and did not have such restrictions.

I also hope your language is capable of handling text based input and output. That's how the engine works currently, check the other starter bots for reference.

I can help you with git once there is something to commit, ping me on IRC.
